src/components/Controls/ColorizeControl/ColorRampRangeSlider.tsx

```diff
@@ -0,0 +1,189 @@
+import { LockOutlined, UnlockOutlined } from "@ant-design/icons";
+import { Tooltip } from "antd";
+import React, { ReactElement, useCallback, useMemo } from "react";
+import { inverseLerp } from "three/src/math/MathUtils";
+
+import { ColorRamp, ColorRampType, isThresholdNumeric } from "../../../colorizer";
+import { thresholdMatchFinder } from "../../../colorizer/utils/data_utils";
+import { useViewerStateStore } from "../../../state";
+import { FlexRowAlignCenter } from "../../../styles/utils";
+
+import { FeatureType } from "../../../colorizer/Dataset";
+import IconButton from "../../IconButton";
+import LabeledSlider from "../../Inputs/LabeledSlider";
+
+type ColorRampRangeSliderProps = {
+  disabled: boolean;
+};
+
+/**
+ * Creates a CSS gradient string for the slider track, between the two range
+ * handles. `min` and `max` are the current range values, while `minRange` and
+ * `maxRange` are the overall bounds of the slider.
+ *
+ * @returns A CSS gradient string, where 0% is at the slider's minimum handle
+ * and 100% is at the slider's maximum handle. When `min` and/or `max` is
+ * outside the slider's bounds, the gradient is extrapolated beyond 0% and/or
+ * 100%.
+ */
+function getTrackCssGradient(
+  min: number,
+  max: number,
+  minRange: number,
+  maxRange: number,
+  colorRamp: ColorRamp
+): string {
+  // Determine the step size between color stops, in percentage space.
+  const range = max - min;
```

Comment:
The step calculation below assumes a non-zero range, is there a guard for `max === min` case that i might've missed?